allennlp/data/image_loader.py

```diff
@@ -37,20 +35,51 @@ def load(self, filenames: ManyPaths) -> List[FloatTensor]:
 class DetectronImageLoader(ImageLoader):
     def __init__(
         self,
-        builtin_config_file: Optional[str] = None,
-        yaml_config_file: Optional[str] = None,
-        overrides: Optional[Dict[str, Any]] = None,
+        pixel_mean: List[float] = [103.530, 116.280, 123.675],
+        pixel_std: List[float] = [1.0, 1.0, 1.0],
+        min_size_train: Tuple[int] = (800,),
+        min_size_train_sampling: str = "choice",
+        max_size_train: int = 1333,
+        min_size_test: int = 800,
+        max_size_test: int = 1333,
+        crop_enabled: bool = False, 
+        crop_type: str = "relative_range",
+        crop_size: List[float] = [0.9, 0.9],
+        image_format: str = "BGR",
+        mask_format: str = "polygon",
     ):
-        from allennlp.common.detectron import get_detectron_cfg
-
-        cfg = get_detectron_cfg(builtin_config_file, yaml_config_file, overrides)
 
+        # constructing the overrides to the cfg file.
+        overrides = {
+            "MODEL": {
+                "PIXEL_MEAN": pixel_mean,
+                "PIXEL_STD": pixel_std,
+            },
+            "INPUT": {
+                "MIN_SIZE_TRAIN": min_size_train,
+                "MIN_SIZE_TRAIN_SAMPLING": min_size_train_sampling,
+                "MAX_SIZE_TRAIN": max_size_train,
+                "MIN_SIZE_TEST": min_size_test,
+                "MAX_SIZE_TEST": max_size_test,
+                "FORMAT": image_format,
+                "MASK_FORMAT": mask_format,
+                "CROP": {
+                    "ENABLED": crop_enabled,
+                    "TYPE": crop_type,
+                    "SIZE": crop_size,
+                },
+            },
+        }
+        from allennlp.common.detectron import get_detectron_cfg
+        cfg = get_detectron_cfg(None, None, overrides)
         from detectron2.data import DatasetMapper
-
         self.mapper = DatasetMapper(cfg)
 
     def load(self, filenames: ManyPaths) -> List[FloatTensor]:
         images = [{"file_name": str(f)} for f in filenames]
+        # TODO: is this efficient enough, the detectron2 impelemntation ? 
         images = [self.mapper(i) for i in images]
-        images = [i["image"] for i in images]
+
+        # TODO: for detectronImage loader, do we want to return image tensor or orginial detectron dict format?  
+        # images = [i["image"] for i in images]
```

Comment:
I wanted to return tensors, so the output of this can be fed straight into `Image2ImageModule` and `ProposalGenerator`. Also, any other implementation of `ImageLoader` would have to return images in Detectron format.

Why would you want to change it? One thing that we don't handle well here is the extra fields for size that Detectron keeps in the dict.